app/models/student_task.rb

```
@@ -1,50 +1,162 @@
+      stage_deadline:     parse_stage_deadline(next_due ? next_due.due_at.in_time_zone(participant.user.try(:timezonepref) || 'UTC') : 'Finished'),
+      permission_granted: participant.permission_granted,
+      participant:        participant
+    )
+    task.revise = task.revise?
+    task.not_started = task.not_started?
+    task
+  end
+
+
+  # create an array of StudentTask instances for all participants linked to a user, sorted by deadline.
+  def self.from_user(user)
```

RuboCop (1.87.0)

[convention] Expected 1 empty line between method definitions; found 2.

(Layout/EmptyLineBetweenDefs)